src/utils/dart.ts

```diff
@@ -14,10 +15,16 @@ export function getDartSdkVersion(output: string): SemanticVersion|undefined {
     return new SemanticVersion(result[0]);
 }
 
-export function checkIfDartIsInstalled(): boolean {
+export async function checkIfDartIsInstalled(): Promise<boolean> {
     const output = execSync("dart --version");
     const version = getDartSdkVersion(output.toString());
 
+    try {
+        await which('dartaotruntime');
+    } catch (e) {
+        throw new Error(GENEZIO_DARTAOTRUNTIME_NOT_FOUND);
```

Comment:
This is thrown here but is not caught in the upper level. This causes an ugly error:
```
[2023-05-03T12:01:12.696Z] DEBUG: An error has occured Error: We couldn't find the "dartaotruntime" executable. Make sure it is probably installed. Check the Dart documentation for more information: https://dart.dev/get-dart.
    at /home/andreia/geneziodev/genezio/build/utils/dart.js:37:19
    at Generator.throw (<anonymous>)
    at rejected (/home/andreia/geneziodev/genezio/build/utils/dart.js:6:65)
[2023-05-03T12:01:12.697Z] DEBUG: An error occured Error: We couldn't find the "dartaotruntime" executable. Make sure it is probably installed. Check the Dart documentation for more information: https://dart.dev/get-dart.
    at /home/andreia/geneziodev/genezio/build/utils/dart.js:37:19
    at Generator.throw (<anonymous>)
    at rejected (/home/andreia/geneziodev/genezio/build/utils/dart.js:6:65) {
  path: './lib/task.dart'
}
We couldn't find the "dartaotruntime" executable. Make sure it is probably installed. Check the Dart documentation for more information: https://dart.dev/get-dart.
```

I think there are 3 places where this should be caught and thrown again, hopefully, it will show the error nicely afterwards:
https://github.com/Genez-io/genezio/blob/dev/src/bundlers/dart/dartBundler.ts#L214
https://github.com/Genez-io/genezio/blob/dev/src/bundlers/dart/localDartBundler.ts#L109
https://github.com/Genez-io/genezio/blob/dev/src/generateSdk/astGenerator/DartAstGenerator.ts#L135